### PR TITLE
Feature | PEO-801 | [Automation workshop] Rules of Rspec

### PR DESCRIPTION
**Pull request by alexyndr**

### CODE REVIEW

File: slides/rules-of-rspec.markdown [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+```ruby
+RSpec.configure do |config|
+  config.before(:each, type: :system) do
+    driven_by :rack_test
+  end
+
+  config.before(:each, type: :system, js: true) do
+    driven_by :selenium_chrome_headless
+  end
+end
```
Comment (rostikkkk2) [inline slides/rules-of-rspec.markdown]: Are you sure that it is default capybara config for db_clean? And may be describe what is `type: :system, js: true`, what is `driven_by :rack_test` and `selenium_chrome_headless`?

File: slides/rules-of-rspec.markdown [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+--
+
+### Data created in before(`:example`) are **rolled back**
+
+Any data you create in a before(`:example` same as `:each`) hook will be rolled back at the end of
```
Comment (rostikkkk2) [inline slides/rules-of-rspec.markdown]: rolled back at the end of the example when use `before`? are you sure?

File: slides/rules-of-rspec.markdown [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    expect(@widget).to do_something
+  end
+
+  it "does something else" do
+    expect(@widget).to do_something_else
+  end
+end
+```
+
+The @widget is recreated in each of the two examples above, so each example
```
Comment (rostikkkk2) [inline slides/rules-of-rspec.markdown]: add ' ' to @widget

File: slides/rules-of-rspec.markdown [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+has a different object, and the underlying data is rolled back so the data
+backing the @widget in each example is new.
+
+`Note` Data created in before(`:context` same as `:all`) are **not rolled back**
```
Comment (rostikkkk2) [inline slides/rules-of-rspec.markdown]: what is same as :all?

File: slides/rules-of-rspec.markdown [hunk 5]
```diff
@@ -0,0 +1,246 @@
+---
+layout: slide
+title: Rules of Rspec
+---
+
+## `let`
```
[the remark this answers is not in the document]
Comment (rostikkkk2) [reply slides/rules-of-rspec.markdown]: and add under title mini guideline about this presentation such as:
1. let and !let
2. hooks before after around
3. db_cleaner

File: slides/rules-of-rspec.markdown [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+### Hooks types
+
+- **after**
+
+  ```ruby
+  after(*args, &block)
+  ```
+
+  Declare a block of code to be run after each example (using `:example`) or once after all examples in the context (using `:context`).
```
[the remark this answers is not in the document]
Comment (alexyndr) [reply slides/rules-of-rspec.markdown]: On the next slide. You can see examples for all types of hooks.

File: slides/rules-of-rspec.markdown (hunk 2 above)
Comment (alexyndr) [reply slides/rules-of-rspec.markdown]: Yes. Rspec documentation https://relishapp.com/rspec/rspec-rails/docs/transactions

File: slides/rules-of-rspec.markdown [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+spec/rails_helper.rb <!-- .element: class="filename" -->
+
+```ruby
+config.use_transactional_fixtures = true
+```
+
+Create a `basic_configure`, which manages the Capybara driver configuration:
+
+spec/support/basic_configure.rb <!-- .element: class="filename" -->
```
[the remark this answers is not in the document]
Comment (alexyndr) [reply slides/rules-of-rspec.markdown]: You right better will be "database_cleaning_configure"

**Pull request merged**